creusot/src/translation/function/promoted.rs

```diff
@@ -24,13 +23,13 @@ pub(crate) fn promoted_signature<'tcx>(body: &Body<'tcx>) -> PreSignature<'tcx>
 
 pub(crate) fn translate_promoted<'tcx>(
     ctx: &mut TranslationCtx<'tcx>,
-    body: &Body<'tcx>,
-    parent: DefId,
+    body_id: BodyId,
 ) -> CreusotResult<(PreSignature<'tcx>, fmir::Body<'tcx>)> {
-    let func_translator = BodyTranslator::build_context(ctx.tcx, ctx, &body, parent);
+    let body = ctx.body(body_id).clone();
```

Comment:
I wish we didn't have to clone the body here but if that turns out to be a perf issue later, we can always either wrap it in an `Rc` or use interior mutability in `ctx` like rustc does.